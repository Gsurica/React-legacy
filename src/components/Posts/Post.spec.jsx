const { render, screen } = require("@testing-library/react")
import { Posts } from './index' 

const props = {
  posts: [
    {
      id: 1,
      title: 'title 1',
      body: 'body 1', 
      cover: 'testing image 1'
    }, 
    {
      id: 2,
      title: 'title 2',
      body: 'body 2', 
      cover: 'testing image 2'
    }, 
    {
      id: 3,
      title: 'title 3',
      body: 'body 3', 
      cover: 'testing image 3'
    }, 
  ]
}

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts { ...props } />)

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3)
    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3)
    expect(screen.getAllByText(/body/i)).toHaveLength(3)
    expect(screen.getByRole('img', { name: /title 3/i })).toHaveAttribute('src', 'testing image 3')
  }) 

  it('should render posts empty', () => {
    render(<Posts />)

    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument()
  })

  it('should match the posts snapshot', () => {
    const { container } = render(<Posts {...props} />) 
    expect(container.firstChild).toMatchSnapshot()
  })

})