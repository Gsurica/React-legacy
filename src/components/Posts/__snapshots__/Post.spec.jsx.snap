// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match the posts snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="testing image 1"
    />
    <div
      class="post-card"
    >
      <h2>
        title 1
      </h2>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="testing image 2"
    />
    <div
      class="post-card"
    >
      <h2>
        title 2
      </h2>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="testing image 3"
    />
    <div
      class="post-card"
    >
      <h2>
        title 3
      </h2>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
